pkg/server/server.go

```diff
@@ -42,8 +42,10 @@ func NewServer(host string, port int, insecureSkipTLS bool, backends ...string)
 	// setting middleware before routes, otherwise it does not work (gin bug)
 	r.Use(ginzap.GinzapWithConfig(zapLogger, &ginzap.Config{}))
 	r.Use(ginzap.RecoveryWithZap(zapLogger, true))
-	r.GET("/", handleHealthz)
-	r.POST("/", sprayProxy.HandleProxy)
+	r.GET("/proxy", handleHealthz)
+	r.POST("/proxy", sprayProxy.HandleProxy)
+	r.POST("/register", sprayProxy.RegisterBackends)
+	r.GET("/unregister", sprayProxy.UnregisterBackends)
```

Comment:
Two things:

1. These endpoints should not be added if `enableDynamicBackends` is false. Let gin take care of sending an appropriate response.
2. I would try to implement a RESTful API scheme here, as such:

```
r.GET("/backends", sprayproxy.GetBackends())
r.POST("/backends", sprayproxy.RegisterBackend())
r.DELETE("/backends/:backend", sprayproxy.UnregisterBackend())
```

OR

```
r.GET("/backends", sprayproxy.GetBackends())
r.POST("/backends", sprayproxy.RegisterBackend())
r.POST("/backends/unregister", sprayproxy.UnregisterBackend())
```